### ISSUE TITLE
SMTP error due to long line-lengths (> 998 octets) in attached logs/patches

### ISSUE THREAD

**Issue by dunderpate** [buildbot/buildbot#5282]
Hi everyone,

I recently started noticing errors in our logs due to long line lengths in the log files that were being attached to emails. I checked and confirmed that the error we are receiving from our SMTP server was in fact correct, and that the MIME contents were in violating RFC 5322 2.1.1. The obvious thing to do here, I think, is to simply base64 encode all logs/patches to avoid this issue and ensure buildbot sends RFC compliant attachments.

So, I went ahead and put together a patch for this and tried it locally with success. I'm working on putting together a PR for the patch and will update the unit tests once I run the patch through the CI system.

```
2020-05-05 18:21:08+0000 [-] Unhandled error in Deferred:
2020-05-05 18:21:08+0000 [-] Unhandled Error
        Traceback (most recent call last):
        --- <exception caught here> ---
          File "/usr/lib/python3.6/site-packages/buildbot/reporters/notifier.py", line 279, in buildMessage
            list(users), patches, logs)
          File "/usr/lib/python3.6/site-packages/buildbot/reporters/mail.py", line 273, in sendMessage
            yield self.sendMail(m, all_recipients)
        twisted.mail._except.SMTPDeliveryError: 550 b'maximum allowed line length is 998 octets, got 8954'
        >>> RCPT TO:<user@domain.tld>
        <<< 250 Accepted
        >>> DATA
        <<< 354 Enter message, ending with "." on a line by itself
        >>> .
        <<< 550 maximum allowed line length is 998 octets, got 8954
```

Thanks everyone,

Dunderpate

**Comment by dunderpate** [comment]
Created PR 5283.

**Comment by dunderpate** [comment]
Wroops, hit the wrong button there :D

**Comment by dunderpate** [comment]
The PR for this one is ready for review now. Feel free to let me know if there is anything that needs to be changed 👍

**Comment by p12tic** [comment]
Fixed in #5283.

**Comment by dunderpate** [comment]
Thanks @p12tic!
